nixos/modules/services/development/fossil-server.nix

```diff
@@ -0,0 +1,146 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.fossil;
+in
+{
+
+  ###### interface
+
+  options.services.fossil = {
+
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+          Enable the Fossil server, which will start an HTTP server to
+          publish Fossil repositories.
+        '';
+    };
+
+    repository = mkOption {
+      type = types.str;
+      default = "/srv/fossil";
+      example = "/fossils/repo.fossil";
+      description = ''
+          The REPOSITORY can be a directory that contains one or more
+          repositories with names ending in ".fossil" or a single file.
+        '';
+    };
+
+    baseurl = mkOption {
+      type = types.str;
+      default = "";
+      example = "example.com";
+      description = "Use URL as the base (useful for reverse proxies)";
+    };
+
+    extroot = mkOption {
+      type = types.str;
+      default = "";
+      description = "Document root for the /ext extension mechanism";
+    };
+
+    files = mkOption {
+      type = types.str;
+      default = "";
+      description = "Comma-separated list of glob patterns for static files";
+    };
+
+    https = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+          Indicates that the input is coming through a reverse proxy
+          that has already translated HTTPS into HTTP.
+        '';
+    };
+
+    jsmode = mkOption {
+      type = types.str;
+      default = "";
```

Comment:
`types.enum` is what you're looking for here.

Reply:
That's much cleaner, thanks - fixed

I noticed that the docstring does not keep newlines in `man configuration.nix`:
![image](https://user-images.githubusercontent.com/15697697/97781211-1499e980-1b8a-11eb-9ac5-f93fe5c8fe41.png)

Is this normal? How could I fix it?